static/js/narrow.js

```diff
@@ -1053,4 +1053,24 @@ exports.hide_empty_narrow_message = function () {
     $("#left_bar_compose_reply_button_big").prop("disabled", false);
 };
 
+exports.allowed_web_public_narrows = [
+    "streams",
+    "stream",
+    "topic",
+    "sender",
+    "has",
+    "search",
+    "near",
+    "id",
+];
+
+exports.is_web_public_compatible = function (ops) {
+    for (const op of ops) {
+        if (!exports.allowed_web_public_narrows.includes(op.operator)) {
+            return false;
+        }
+    }
+    return true;
+};
```

Comment:
Can you add frontend tests for this?  Might belong in the `filter.js` suite (and possible this function belongs in that non-test module)?

Reply:
Okay, let me try to do that.